src/internal/components/configurationControls/PushpinSizeControl.tsx

```diff
@@ -0,0 +1,54 @@
+// (C) 2019-2020 GoodData Corporation
+import * as React from "react";
+import { WrappedComponentProps, injectIntl } from "react-intl";
+
+import get = require("lodash/get");
+import { IVisualizationProperties } from "../../interfaces/Visualization";
+import ConfigSubsection from "./ConfigSubsection";
+import DropdownControl from "./DropdownControl";
+import { getTranslatedDropdownItems } from "../../utils/translations";
+import { pushpinSizeDropdownItems } from "../../constants/dropdowns";
+
+export interface IPushpinSizeControl {
+    disabled: boolean;
+    properties: IVisualizationProperties;
+    pushData: (data: any) => any;
+}
+
+class PushpinSizeControl extends React.PureComponent<IPushpinSizeControl & WrappedComponentProps, {}> {
```

Comment:
we can remove the empty type, right?

Reply:
It could be functional component
```

function getPushpinProperty(props: IPushpinSizeControl & WrappedComponentProps) {
        return get(props, "properties.controls.pushpin", {});
    }

function PushpinSizeControl(props: IPushpinSizeControl & WrappedComponentProps): React.ReactElement {
    const { minSize = "default", maxSize = "default" } = getPushpinProperty(props);
    const { disabled, properties, pushData, intl } = props;
    return (
        <ConfigSubsection title="properties.points.size.title">
            <DropdownControl
                value={minSize}
                valuePath="pushpin.minSize"
                labelText="properties.points.size.min.title"
                disabled={disabled}
                showDisabledMessage={disabled}
                properties={properties}
                pushData={pushData}
                items={getTranslatedDropdownItems(pushpinSizeDropdownItems, intl)}
            />
            <DropdownControl
                value={maxSize}
                valuePath="pushpin.maxSize"
                labelText="properties.points.size.max.title"
                disabled={disabled}
                showDisabledMessage={disabled}
                properties={properties}
                pushData={pushData}
                items={getTranslatedDropdownItems(pushpinSizeDropdownItems, intl)}
            />
        </ConfigSubsection>
    );
}
```